### PR TITLE
feat: add fmt, test and clippy to CI

### PR DESCRIPTION
**Pull request by juanbono**
**Motivation**

This PR adds a step for `cargo clippy` and one for `cargo fmt`

**Description**

Closes #issue_number

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -31,7 +32,19 @@ jobs:
       - name: Run cargo build
         run: |
           cargo build
-  
+      
+      - name: Run cargo clippy
+        run: |
+          cargo clippy --all-targets --all-features -- -D warnings
```
Comment (MegaRedHand) [inline .github/workflows/ci.yaml:38]: Shouldn't we add `--workspace` too?

Comment (juanbono) [reply .github/workflows/ci.yaml:38]: thanks!

Comment (juanbono) [reply .github/workflows/ci.yaml]: done

**Review by MegaRedHand — APPROVED** [review]
🦭👍

**Pull request merged**